src/app/admin/modules/data-management/components/visual-graph-builder/visual-graph-builder.component.ts

```diff
@@ -0,0 +1,222 @@
+import { Component, OnInit , Input} from '@angular/core';
+import * as echarts from 'echarts';
+import { EChartOption } from 'echarts';
+import { GraphData } from 'src/app/shared/interfaces/graph-data';
+import { PieGraphData } from 'src/app/shared/interfaces/pie-graph-data';
+import { ParamSelector } from 'src/app/shared/interfaces/visual-graph-param-selector';
+
+@Component({
+  selector: 'app-visual-graph-builder',
+  templateUrl: './visual-graph-builder.component.html',
+  styleUrls: ['./visual-graph-builder.component.css']
+})
+
+export class VisualGraphBuilderComponent implements OnInit {
+
+  @Input() titleGraphNames: string;
+  @Input() isPieGraph: boolean;
+
+  chartOption?: EChartOption;
+  echartsInstance: echarts.ECharts;
+  pieGraphData: PieGraphData[] = [];
+  title: string;
```

Comment:
If `title` is not used in your HTML, I see no reasons to keep it as a field of the component.